spec/models/event_stream_spec.rb

```diff
@@ -0,0 +1,28 @@
+def with_event_types
+  EventStream.event_groups.each do |group_name, group_data|
+    EventStream::GROUP_LEVELS.each do |level|
+      group_data[level]&.each { |typ| yield(group_name, level, typ) }
+    end
+  end
+end
+
+describe EventStream do
+  describe ".event_groups" do
+    with_event_types do |group, level, typ|
+      it ":#{group}/:#{level}/#{typ} is string or regex" do
+        expect(typ.kind_of?(Regexp) || typ.kind_of?(String)).to eq(true)
+      end
+
+      if typ.kind_of?(Regexp)
+        it ":#{group}/:#{level}/#{typ} is usable in SQL queries" do
+          expect { EventStream.where("event_type ~ ?", typ.source).to_a }
```

Comment:
Should we add the `typ.casefold? ? "event_type ~ ?" : "event_type ~*"` case to match what being done in timeline?

Reply:
I did not do this because validity of the regular expression and its case (in)sensitivity are independent and do not affect each other (invalid regular expression cannot be made valid by using different match operator). And since we only care about validity here, I decided to keep things as simple as possible.